clarity/src/vm/analysis/type_checker/mod.rs

```diff
@@ -755,13 +1147,117 @@ impl<'a, 'b> TypeChecker<'a, 'b> {
             }
         };
 
+        analysis_typecheck_cost(self, expected_type, &expr_type)?;
+        match (expected_type, &expr.expr) {
+            (
+                TypeSignature::CallableType(CallableSubtype::Trait(expected_trait_id)),
+                LiteralValue(Value::Principal(PrincipalData::Contract(ref contract_identifier))),
+            ) => {
+                // When a principal literal is used as a trait, make sure it implements the trait.
+                let contract_to_check = self
+                    .db
+                    .load_contract(&contract_identifier)
+                    .ok_or(CheckErrors::NoSuchContract(contract_identifier.to_string()))?;
+                runtime_cost(
+                    ClarityCostFunction::AnalysisFetchContractEntry,
+                    &mut self.cost_track,
+                    1,
```

Comment:
The cost here should be the length of the contract code body.

Reply:
I'm now just using the length of the public and read-only functions as a proxy for this size. The contract code body, or its size, are not actually stored here.